### DISCUSSION TITLE
221212 Meeting ideas

### DISCUSSION THREAD

**Discussion by ehhartmann** [graeter-group/kimmdy#80] [Ideas]
# Topology and Parameter Patch Refactor
- 

# Detailed Balance update
- " It states that at equilibrium, each elementary process is in equilibrium with its reverse process."
- Roux paper used symmetric momentum reversal for MD/MC simulations to ensure detailed balance
- momentum reversal was applied to short neMD simulations which were used to calculate ΔE of a reaction
 -> we already have ΔE for HAT and it's reversible, homolysis is seen as irreversible at the moment (which is close to the truth?!)
-> unclear how this would apply to rejection free kMC
-> reaction MD is only used to get product state, no meaning associated to it
- do we want to stay in detailed balance? Can we regain detailed balance?
-> view time before reaction as in equilibrium, reaction MD + first part of next pulling MD as transformation to the next equilibrium --> in  detailed balance
 
# Using KIMMDY
- which features should be part of KIMMDY v2?
    - better (espaloma) parameters? partial charges?
    - documentation? 
- what do we want to use KIMMDY v2 for?
- do we immediately want to continue developing KIMMDY after v2 release? 
- which features do we want to implement next?

# Open questions from last meeting
- How to get prefactor? QM/MM? pure QM?
- MC w/ rejection:
   - cutoff in time domain
 * How to mix reactions?
	* option 1: Only do one kind at a time
	* option 2: Competing rates need to be evaluated at the same time (as we currently do)
	* Competing rates can have diff prefactors
* maybe recalculate prefactor for homolysis in QMMM together with HAT QMMM

**Comment by KRiedmiller** [comment]
Protocol: 
- Topology changes  
	- goal: general solution for changing topology  
	- in memory representation  
	- Testing in place  
	- Todo: regression tests with new KIMMDY versions  
	- Bonds can be broken and rebuild from atom index  
	- Atom types get changed using patches -> parameters get changed by factors or absolute values  
	- Todo: charges must add up after applying patch!  
	- Todo: Check /# in FF  
	- How to store radical? Change atom name  
		- gmx might complain  
		- usefull for analysis  
- Detailed Balance update  
	- Momentum reversal:  
		- Each process in eq w/ iths reversed process  
		- To ensure this, vel reversed  
	- Not important for practical application  
	- MC step starts from eq, ends in eq, should be in detailed balance all the time, exluding the reaction itself  
	- Maybe just try  
- Using KIMMDY/Release  
	- [x] Partial charges should be good; they are!  
	- [x] Espaloma parameters as new verions; already implemented!  
	- Publishing:  
		- Methods  
		- one use case / example like hexala for statistics and details  
		- one big one to show off  
		- maybe use ala-ala-dopa-ala-ala  
			- would be specifically for collagen  
	- Documentation  
		- [ ] example  
		- [x] auto generated technical documentation  
	- Features:  
		- Espaloma -> next paper  
		- Collagen -> next paper  
		- MC w/ rejection + testing -> first release!  
- Open Questions  
	- Prefactor:  
		- QM/MM  
		- Ab initio MD: g09 issue: frozen part, harmonic restrains needed  
		- calculating for homolysis and HAT together makes sense  
	- Topology changer as standalone tool?  
		- for MDAnalysis  
		- for gromacs  
		- would be alot of work  
		- for now keep it inside KIMMDY
